### ISSUE TITLE
Issue: How to use other natural languages besides English?

### ISSUE THREAD

**Issue by f3rn4nd0-c354r** [langchain-ai/langchain#13250]
### Issue you'd like to raise.

Hello,
Thanks for this framework. It's making everyone's work simpler!

I'm using an LLM model to infer data from Portuguese websites, and expecting answers in Portuguese. But some of Langchain features, namely the "Get format instructions" for Output Parses come written in English. I'm not sure what the best approach would be here.

Should I do all my prompting in English, and just add "Answer in Portuguese" at the end?

Should this be a feature in Langchain, and if so, how would it work?

I'm not sure asking the community to translate it would be the right approach, because this is not just about translation, but making sure the prompts work correctly. I would be fine with some api call to replace the English text with my translation, but that doesn't seem to be part of the public API at the moment.

Thanks,

**Comment by f3rn4nd0-c354r** [comment]
Thanks Dosubot.

I was looking for some guidance. Is anyone working with other natural languages besides English? What do you recomend? Prompting in English and adding a request to have an answer in other language?

Is LangChain planning to be international friendly, by providing easy methods to provide translated prompts? I can help with this, if that's the way.

**Comment by davivc** [comment]
Hello, @f3rn4nd0-c354r. I don't know if you already solve your issue but I've been using some langchain tools and until now I didn't find any problem mixing Portuguese with English instructions. I think this case can be solved with prompt engineering. I gave explicit instructions in the system message to answer only in Portuguese and that the assistant don't know other languages. Also for the human response I reinforce that the assistant must answer in Portuguese. As of today, I've tested with OpenAI 4, Gemini 1.5 Flash and Mistral 7x13B and it worked well with these 3. I'm using tool calling with create_json_chat_agent and AgentExecutor.

I'm still improving our company's implementation of this agent, so if I find myself struggling with this problem I'll come back here to let you know.
